src/mlpack/core/cv/simple_cv.hpp

```diff
@@ -0,0 +1,131 @@
+/**
+ * @file simple_cv.hpp
+ * @author Kirill Mishchenko
+ *
+ * A simple cross-validation strategy.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_CORE_CV_SIMPLE_CV_HPP
+#define MLPACK_CORE_CV_SIMPLE_CV_HPP
+
+#include <mlpack/core/cv/meta_info_extractor.hpp>
+#include <mlpack/core/cv/cv_base.hpp>
+
+namespace mlpack {
+namespace cv {
+
+/**
+ * The class SimpleCV splits data into training and validation sets, runs
+ * training on the training set and evaluates performance on the validation set.
+ *
+ * @tparam MLAlgorithm A machine learning algorithm.
+ * @tparam Metric A metric to assess the quality of a trained model.
+ * @tparam MatType The type of data.
+ * @tparam PredictionsType The type of predictions (should be passed when the
+ *     predictions type is a template parameter in Train methods of
+ *     MLAlgorithm).
+ * @tparam WeightsType The type of weights (should be passed when weighted
+ *     learning is supported, and the weights type is a template parameter in
+ *     Train methods of MLAlgorithm).
+ */
+template<typename MLAlgorithm,
+         typename Metric,
+         typename MatType = arma::mat,
+         typename PredictionsType =
+             typename MetaInfoExtractor<MLAlgorithm, MatType>::PredictionsType,
+         typename WeightsType =
+             typename MetaInfoExtractor<MLAlgorithm, MatType,
+                 PredictionsType>::WeightsType>
+class SimpleCV :
+    private CVBase<MLAlgorithm, MatType, PredictionsType, WeightsType>
+{
+ public:
+  /**
+   * This constructor splits data into training and validation sets.
+   *
+   * @param validationSize A proportion (between 0 and 1) of the data used as a
+   *     validation set.
+   * @param args Basic constructor arguments for MLAlgortithm (see the CVBase
+   *     constructors for reference).
+   */
+  template<typename... CVBaseArgs>
+  SimpleCV(const float validationSize, CVBaseArgs...args);
+
+  /**
+   * Train on the training set and assess performance on the validation set by
+   * using the class Metric.
+   *
+   * @param args Arguments for MLAlgorithm (in addition to the passed
+   *     ones in the constructor).
+   */
+  template<typename... MLAlgorithmArgs>
+  double Evaluate(const MLAlgorithmArgs& ...args);
```

Comment:
Right, so my understanding here is that we are not passing `MatType`, `LabelsType`, or `numClasses` to `Evaluate()` but instead to the `SimpleCV` constructor.  In either case, the question may be better stated like this--- would it be better to make the `MatType`, `LabelsType`, `numClasses`, and/or `WeightsType` parameters explicit, instead of wrapped into the `CVBaseArgs` parameter pack?  I think it would make the class easier to understand for users, but I understand it could make the implementation tedious.

If you don't want to do that, then we should be sure to make the documentation really clear with examples so that users know how to use the class.  I don't think they will be able to piece it together from what we have now, but a simple walkthrough example (or two) could get most people what they need.